### PR TITLE
add configuration option to restrict users by default

### LINKED ISSUE #427: User with limited permissions when self registered
Hi,

we plan to go live with Gitea for a bigger project where we expect to have many user. The current problem we face is that when a user register (what he should do) he can create as example organisations, or can create own repos. 

We need that user can register but by default they should be only able to create issues. Otherwise our platform will end up to something like Github. We expect to have more then 1000 public repos. 

We plan to go live soon but with the current restrictions it will simple mot be possible to restrict users in what the able to do.

What is missing from our point of view is a user role as "reporter". Later an admin can grant more right that they can create own repo a example. 

Would be nice this could be implemented in Gitea. 

Thank you 
Reiner

### PR DESCRIPTION
**Pull request by rjnienaber**
Resolves #427.

Adds a configuration option to **app.ini** that makes new users have the `IsRestricted` permission set to `true` by default.

### CODE REVIEW

File: integrations/sqlite.ini.tmpl [hunk 1]
```diff
@@ -62,6 +62,7 @@ ENABLE_CAPTCHA                    = false
 REQUIRE_SIGNIN_VIEW               = false
 DEFAULT_KEEP_EMAIL_PRIVATE        = false
 DEFAULT_ALLOW_CREATE_ORGANIZATION = true
+DEFAULT_USER_IS_RESTRICTED 				= false
```
Comment (adelowo) [inline integrations/sqlite.ini.tmpl]: Doesn't seem properly aligned

Comment (rjnienaber) [reply integrations/sqlite.ini.tmpl]: Thanks. My IDE was hiding this from me.

**Review by adelowo — APPROVED** [review]

File: custom/conf/app.example.ini [hunk 2]
```diff
@@ -649,6 +649,9 @@ PATH =
 ;; Default value for AllowCreateOrganization
 ;; Every new user will have rights set to create organizations depending on this setting
 ;DEFAULT_ALLOW_CREATE_ORGANIZATION = true
+;; Default value for IsRestricted
+;; Every new user will have restricted permissions depending on this setting
+;DEFAULT_USER_IS_RESTRICTED = true
```
Comment (6543) [inline custom/conf/app.example.ini]:
```suggestion
;DEFAULT_USER_IS_RESTRICTED = false
```

File: integrations/mssql.ini.tmpl [hunk 3]
```diff
@@ -65,6 +65,7 @@ ENABLE_CAPTCHA                    = false
 REQUIRE_SIGNIN_VIEW               = false
 DEFAULT_KEEP_EMAIL_PRIVATE        = false
 DEFAULT_ALLOW_CREATE_ORGANIZATION = true
+DEFAULT_USER_IS_RESTRICTED 				= false
```
Comment (6543) [inline integrations/mssql.ini.tmpl]:
```suggestion
```

File: integrations/mysql.ini.tmpl [hunk 4]
```diff
@@ -85,6 +85,7 @@ ENABLE_CAPTCHA                    = false
 REQUIRE_SIGNIN_VIEW               = false
 DEFAULT_KEEP_EMAIL_PRIVATE        = false
 DEFAULT_ALLOW_CREATE_ORGANIZATION = true
+DEFAULT_USER_IS_RESTRICTED 				= false
```
Comment (6543) [inline integrations/mysql.ini.tmpl]:
```suggestion
```

File: integrations/mysql8.ini.tmpl [hunk 5]
```diff
@@ -63,6 +63,7 @@ ENABLE_CAPTCHA                    = false
 REQUIRE_SIGNIN_VIEW               = false
 DEFAULT_KEEP_EMAIL_PRIVATE        = false
 DEFAULT_ALLOW_CREATE_ORGANIZATION = true
+DEFAULT_USER_IS_RESTRICTED 				= false
```
Comment (6543) [inline integrations/mysql8.ini.tmpl]:
```suggestion
```

File: integrations/pgsql.ini.tmpl [hunk 6]
```diff
@@ -66,6 +66,7 @@ ENABLE_CAPTCHA                    = false
 REQUIRE_SIGNIN_VIEW               = false
 DEFAULT_KEEP_EMAIL_PRIVATE        = false
 DEFAULT_ALLOW_CREATE_ORGANIZATION = true
+DEFAULT_USER_IS_RESTRICTED 				= false
```
Comment (6543) [inline integrations/pgsql.ini.tmpl]:
```suggestion
```

File: integrations/sqlite.ini.tmpl [hunk 7]
```diff
@@ -62,6 +62,7 @@ ENABLE_CAPTCHA                    = false
 REQUIRE_SIGNIN_VIEW               = false
 DEFAULT_KEEP_EMAIL_PRIVATE        = false
 DEFAULT_ALLOW_CREATE_ORGANIZATION = true
+DEFAULT_USER_IS_RESTRICTED        = false
```
Comment (6543) [inline integrations/sqlite.ini.tmpl]:
```suggestion
```

File: integrations/signup_test.go [hunk 8]
```diff
@@ -33,6 +33,25 @@ func TestSignup(t *testing.T) {
...
+		"user_name": "restrictedUser",
+		"email":     "restrictedUser@example.com",
+		"password":  "examplePassword!1",
+		"retype":    "examplePassword!1",
+	})
+	MakeRequest(t, req, http.StatusFound)
+
+	// should be able to view new user's page
+	req = NewRequest(t, "GET", "/restrictedUser")
+	MakeRequest(t, req, http.StatusOK)
```
Comment (6543) [inline integrations/signup_test.go:53]: TODO: loaf new user from db and check if created user is restricted

File: integrations/signup_test.go [hunk 9]
```diff
@@ -5,6 +5,7 @@
 package integrations
 
 import (
+	"code.gitea.io/gitea/models"
```
Comment (6543) [inline integrations/signup_test.go]: Sort imports

Comment (rjnienaber) [reply integrations/signup_test.go]: Would it be worthwhile adding something like [gci](https://github.com/daixiang0/gci) to the ci workflow to enforce import ordering?

**Review by 6543 — APPROVED** [review]

File: integrations/signup_test.go (hunk 9 above)
Comment (6543) [reply integrations/signup_test.go]: it already does fail ;)

Comment (rjnienaber) [reply integrations/signup_test.go]: Hmm, I'm confused then. It looks to me like [the CI build](https://drone.gitea.io/go-gitea/gitea/41663) succeeded?

Comment (6543) [reply integrations/signup_test.go]: strange but true - perhaps "go vet" do not account testing files

Comment (rjnienaber) [reply integrations/signup_test.go]: That does appear to be [the case](https://github.com/go-gitea/gitea/blob/27c15782352aabaf9279c9c187d3658649d81e2f/Makefile#L96):

```
GO_PACKAGES ?= $(filter-out code.gitea.io/gitea/models/migrations code.gitea.io/gitea/integrations/migration-test code.gitea.io/gitea/integrations,$(shell $(GO) list -mod=vendor ./... | grep -v /vendor/))
```

**Comment by techknowlogick** [comment]
🚀

**Pull request merged**